In [2]:
import pandas as pd

# Датасет о расходах на страховку

Ссылка на датаесет: https://www.kaggle.com/datasets/mirichoi0218/insurance

In [3]:
cost_strax = pd.read_csv('cost_strax.csv')

## Выясним количество признаков и объектов

In [4]:
cost_strax.shape

(1338, 7)

Таким образом в датасете 1338 объектов, каждый из которых характеризуется 7 признаками.

## Посмотрим какие столбцы (будущие признаки) у нас есть.

In [5]:
cost_strax.columns

Index(['age', 'sex', 'bmi', 'children', 'smoker', 'region', 'charges'], dtype='object')

## Опишем столбцы:

age - возраст человека.

sex - пол человека.

bmi - индекс массы тела. Вычисляется как kg / h^2 (h - рост человека в метрах). Норма от 18,5 до 24,9.

children - количество детей, обслуживаемых мед страховкой.

smoker - курит или нет.

region - район проживания в США.

charges - сумма затра по страховке.

Объектом в данном датасете является человек.

Целевым признаком является столбец charges.

Существенными признаками будут все отсальные столбцы (age, sex, bmi, children, smiker, region).

## Посмотрим на данные.

In [6]:
cost_strax.dtypes

age           int64
sex          object
bmi         float64
children      int64
smoker       object
region       object
charges     float64
dtype: object

In [7]:
cost_strax.sample(5)

age     sex    bmi  children smoker     region      charges
1251   19    male  19.80         0     no  southwest   1241.56500
772    44  female  36.48         0     no  northeast  12797.20962
365    49  female  30.78         1     no  northeast   9778.34720
608    28    male  26.98         2     no  northeast   4435.09420
469    18  female  24.09         1     no  southeast   2201.09710

## Определим типы признаков:

Количественные признаки: age, bmi, children, charges.

Бинарные категориальные признаки: sex, smoker.

Категориальные признаки: region.

Данный датасет подходит для задачи регрессии. Предсказывать будем сумму страховки.

# Датасет о стоимости телефона

Ссылка на датаесет: https://www.kaggle.com/datasets/abdelsalam112/phone-price-classification/data

In [6]:
cost_phone_train = pd.read_csv('./cost_phone/train.csv')
cost_phone_test = pd.read_csv('./cost_phone/test.csv')
cost_phone = cost_phone_train + cost_phone_test

## Выясним количество объектов и признаков

In [10]:
cost_phone.shape

(2000, 22)

В данном датасете 2000 объектов и 22 признака.

## Посмотрим какие столбцы (будущие признаки) у нас есть.

In [7]:
cost_phone.columns

Index(['battery_power', 'blue', 'clock_speed', 'dual_sim', 'fc', 'four_g',
       'id', 'int_memory', 'm_dep', 'mobile_wt', 'n_cores', 'pc',
       'price_range', 'px_height', 'px_width', 'ram', 'sc_h', 'sc_w',
       'talk_time', 'three_g', 'touch_screen', 'wifi'],
      dtype='object')

## Опишем столбцы:

battery_power - ёмкость аккумулятора.

blue - наличие bluetooth

clock_speed - частота процессора

dual_sim - поддержка второй сим-карты

fc - разрешение фронтальной камеры

four_g - поддержка 4G

id - просто id

int_memory - объём внутренней памяти

m_dep - толщина телефона в сантиметрах.

mobile_wt - вес телефона

n_cores - число ядер процессора

pc - разрешение основной камеры

price_range - ценовая категория телефона

px_height - количество пикселей по вертикали

px_weight - количество пикселей по горизонтали

ram - объём оперативной памяти

sc_h - высота экрана телефона

sc_w - ширина экрана телефона

talk_time - на сколько хватит аккумулятора при непрерывном разговоре

three_g - поддержка 3G

touch_screen - сенсорный или нет

wifi - поддержка wifi

Объектом в данном датасете является телефон.

Ключевым признаком является price_range.

Существенными признаками являются все остальные признаки, кроме id.

Несущественным признаком является id.

## Посмотрим на данные

In [8]:
cost_phone.dtypes

battery_power    float64
blue             float64
clock_speed      float64
dual_sim         float64
fc               float64
four_g           float64
id               float64
int_memory       float64
m_dep            float64
mobile_wt        float64
n_cores          float64
pc               float64
price_range      float64
px_height        float64
px_width         float64
ram              float64
sc_h             float64
sc_w             float64
talk_time        float64
three_g          float64
touch_screen     float64
wifi             float64
dtype: object

In [9]:
cost_phone.sample(5)

battery_power  blue  clock_speed  dual_sim    fc  four_g  id  \
1218            NaN   NaN          NaN       NaN   NaN     NaN NaN   
1756            NaN   NaN          NaN       NaN   NaN     NaN NaN   
625          2069.0   0.0          4.1       1.0  16.0     1.0 NaN   
218          3584.0   0.0          1.0       0.0   6.0     2.0 NaN   
1010            NaN   NaN          NaN       NaN   NaN     NaN NaN   

      int_memory  m_dep  mobile_wt  ...  price_range  px_height  px_width  \
1218         NaN    NaN        NaN  ...          NaN        NaN       NaN   
1756         NaN    NaN        NaN  ...          NaN        NaN       NaN   
625         76.0    0.4      250.0  ...          NaN     1165.0    2673.0   
218         45.0    0.5      308.0  ...          NaN     1423.0    3212.0   
1010         NaN    NaN        NaN  ...          NaN        NaN       NaN   

         ram  sc_h  sc_w  talk_time  three_g  touch_screen  wifi  
1218     NaN   NaN   NaN        NaN      NaN           NaN   NaN  
1756     NaN   NaN   NaN        NaN      NaN           NaN   NaN  
625   2640.0  26.0  14.0        4.0      1.0           1.0   1.0  
218   1957.0  23.0  10.0       23.0      2.0           1.0   2.0  
1010     NaN   NaN   NaN        NaN      NaN           NaN   NaN  

[5 rows x 22 columns]

## Определим типы признаков

Количественные признаки: battery_power, clock_speed, fc, int_memory, m_dep, mobile_wt, n_cores, pc, px_height, px_width, ram, sc_h, sc_w, talk_time.

Бинарные категориальные признаки: blue, dual_sim, four_g, three_g, touch_screen, wifi.

Категориальные признаки: price_range.

Порядковые признаки: id.

Данный датасет подходит для задачи классификации, будем предсказывать класс стоимости телефона.

# Датасет о студентах

Ссылка на датасет: https://www.kaggle.com/datasets/rkiattisak/student-performance-in-mathematics

In [14]:
student = pd.read_csv('exams.csv')

## Определим количество объектов и признаков

In [15]:
student.shape

(1000, 8)

Количество объектов - 1000, признаков - 8

## Посмотрим какие столбцы (будущие признаки) у нас есть.

In [16]:
student.columns

Index(['gender', 'race/ethnicity', 'parental level of education', 'lunch',
       'test preparation course', 'math score', 'reading score',
       'writing score'],
      dtype='object')

## Опишем столбцы

gender - пол ребёнка.

race/ethnicity - расса или этническая принадлежность.

parental level of education - наивысший уровень образования, достигнутый родителями.

lunch - получает ли студент бесплатный обед.

test preparation - пройден ли курс подготовки к тесту.

math score - оценка за математику.

reading score - оценка за чтение.

writing score - оценка за письмо.

Объектом в данном датасете является студент.

Целевым признаком является math score.

Существенными признаками являются: parental level of education, test preparation.

Несущественными признаками являются оставшиеся признаки.

## Посмотрим на данные

In [17]:
student.dtypes

gender                         object
race/ethnicity                 object
parental level of education    object
lunch                          object
test preparation course        object
math score                      int64
reading score                   int64
writing score                   int64
dtype: object

In [18]:
student.sample(5)

gender race/ethnicity parental level of education         lunch  \
60     male        group D            some high school      standard   
576    male        group D                some college      standard   
498    male        group D          associate's degree      standard   
191  female        group A            some high school      standard   
38     male        group C          associate's degree  free/reduced   

    test preparation course  math score  reading score  writing score  
60                completed          82             74             75  
576                    none          74             74             70  
498                    none          97             94             91  
191               completed          64             81             87  
38                completed          74             85             87

## Определим типы признаков

Количественные признаки: math score, reading score, writeng score.

Бинарные категориальные признаки: gender, lunch, test preparation course.

Категориальные признаки: race/ethnicity, parental level of education.

Данный датасет подходит для задачи регрессии, будем предсказывать оценку по метематике.

# Датасет о состоянии сердца

Ссылка на датасет: https://www.kaggle.com/datasets/rashikrahmanpritom/heart-attack-analysis-prediction-dataset

In [21]:
heart = pd.read_csv('./heart/heart.csv')

## Определим количество объектов и признаков

In [22]:
heart.shape

(303, 14)

Количество объектов - 303, количество признаков - 14.

## Посмотрим какие столбцы (будущие признаки) у нас есть.

In [23]:
heart.columns

Index(['age', 'sex', 'cp', 'trtbps', 'chol', 'fbs', 'restecg', 'thalachh',
       'exng', 'oldpeak', 'slp', 'caa', 'thall', 'output'],
      dtype='object')

## Опишем столбцы

age - возраст человека.

sex - пол человека.

cp - тип боли в груди (всего 4 вида).

trtbps - артериальное давление в состоянии покоя.

chol - уровень холестерина.

fbs - уровень сахара в крови натощак.

restecg - результаты электрокардиографии в состоянии покоя.

thalachn - максимальная достигнутая частота сердечных сокращений.

exng - стенокардия.

oldpeak - пик до этого.

slp - slope.

caa - количество крупных сосудов.

thall - скорость таля.

output - приступ или нет.

Объектом в данном датасете является человек.

Целевым признаков является output.

Существенными признаками являются все остальные признаки.

Несущественных признаков нет.

## Посмотрим на данные

In [24]:
heart.dtypes

age           int64
sex           int64
cp            int64
trtbps        int64
chol          int64
fbs           int64
restecg       int64
thalachh      int64
exng          int64
oldpeak     float64
slp           int64
caa           int64
thall         int64
output        int64
dtype: object

In [25]:
heart.sample(5)

age  sex  cp  trtbps  chol  fbs  restecg  thalachh  exng  oldpeak  slp  \
260   66    0   0     178   228    1        1       165     1      1.0    1   
120   64    0   0     130   303    0        1       122     0      2.0    1   
31    65    1   0     120   177    0        1       140     0      0.4    2   
188   50    1   2     140   233    0        1       163     0      0.6    1   
130   54    0   2     160   201    0        1       163     0      0.0    2   

     caa  thall  output  
260    2      3       0  
120    2      2       1  
31     0      3       1  
188    1      3       0  
130    1      2       1

## Определим типы признаков:

Количественные признаки: age, trtbps, chol, thalach, oldpeak, caa, thall.

Бинарные категориальные признаки: sex, fbs, exng, output.

Категориальные признаки: cp, restecg, slp.

Данный датасет подходит для задачи классификации, будем предсказывать приступ.